skyportal/facility_apis/atlas.py

```diff
@@ -0,0 +1,310 @@
+from astropy.time import Time
+import numpy as np
+import pandas as pd
+from io import StringIO
+
+from . import FollowUpAPI
+from . import FakeApplication, FakeRequest
+from baselayer.app.env import load_env
+
+from ..utils import http
+
+env, cfg = load_env()
+
+if cfg['app.atlas.port'] is None:
+    ATLAS_URL = f"{cfg['app.atlas.protocol']}://{cfg['app.atlas.host']}"
+else:
+    ATLAS_URL = (
+        f"{cfg['app.atlas.protocol']}://{cfg['app.atlas.host']}:{cfg['app.atlas.port']}"
+    )
+
+
+class ATLASRequest:
+
+    """A dictionary structure for ATLAS forced photometry requests."""
+
+    def _build_payload(self, request):
+        """Payload json for ATLAS forced photometry requests.
+
+        Parameters
+        ----------
+
+        request: skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        Returns
+        ----------
+        payload: json
+            payload for requests.
+        """
+
+        mjd_min = Time(request.payload["start_date"], format='iso').mjd
+        mjd_max = Time(request.payload["end_date"], format='iso').mjd
+
+        if mjd_max < mjd_min:
+            raise ValueError('mjd_min must be smaller than mjd_max')
+
+        target = {
+            'ra': request.obj.ra,
+            'dec': request.obj.dec,
+            'mjd_min': mjd_min,
+            'mjd_max': mjd_max,
+            'send_email': False,
+        }
+
+        return target
+
+
+class ATLASAPI(FollowUpAPI):
+
+    """An interface to ATLAS forced photometry."""
+
+    @staticmethod
+    def get(request):
+
+        """Get a forced photometry request result from ATLAS.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+        """
+
+        from ..models import (
+            DBSession,
+            FollowupRequest,
+            FacilityTransaction,
+            Allocation,
+            Instrument,
+        )
+
+        req = (
+            DBSession()
+            .query(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .one()
+        )
+
+        instrument = (
+            Instrument.query_records_accessible_by(request.requester)
+            .join(Allocation)
+            .join(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .first()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        content = req.transactions[0].response["content"]
+        content = json.loads(content)
+
+        r = requests.get(
+            content["url"],
+            headers={
+                'Authorization': f"Token {altdata['api_token']}",
+                'Accept': 'application/json',
+            },
+        )
+        r.raise_for_status()
+
+        if r.status_code == 200:
```

Comment:
Easier to short-circuit here:

```
if r.status_code != 200:
    ... do something ...
```

I don't quite understand the purpose of the if clause as-is; it just seems to set the request status, which never gets used again.